docs/migration/dapp_migration.md

```diff
@@ -37,7 +37,7 @@ For starters, here's a sampling of exciting perks dapps with get with the Nitro
     - Auto-redeem will not be created if the user does not have enough balance to pay for `gasFeeCap * gasLimit` (meaning you can no longer set a max gas fee cap).
     - Deposited gas will be refunded to `excessFeeRefundAddress` if it cannot create an auto-redeem.
     - The user will be refunded the submission cost of their retryable if it is auto-redeemed.
-- **Arbitrum blockhash**: `blockhash(x)` returns Arbitrum blockhash of a specific L1 block `x` if `block.number - 256 <= x < block.number`, otherwise `0`. Arbitrum blockhash is **NOT** the L1 block hash but a value deterministically generated from the L1 block hash and state of the inbox. `blockhash(block.number)` will always return `0` in nitro in-line with ethereum.
+- **Arbitrum blockhash**: `blockhash(x)` returns the block hash of a specific L1 block number `x`, if it is known and within the range `block.number - 256 <= x < block.number`. If `x` is outside of this range, `blockhash(x)` will return `0`. This includes `blockhash(block.number)`, which always returns `0` just like on Ethereum. **Usually the block hash is not known for a given `x`.** In these cases `blockhash(x)` will return a fake but unique block hash deterministically generated from the newest, real hash whenever one is added.
```

Comment:
I think this explanation still needs some work. I think the specific behavior is:
The first time we see a new L1 block, we set the block hash of the previous L1 block to the block hash of the previous L2 block. If there were any L1 blocks passed with no associated L2 blocks, we backfill deterministically using the same L2 block hash